externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkTBinaryFrontendService.scala

```diff
@@ -94,7 +94,11 @@ class SparkTBinaryFrontendService(
   }
 
   override def attributes: Map[String, String] = {
-    Map(KYUUBI_ENGINE_ID -> KyuubiSparkUtil.engineId)
+    var map = Map(KYUUBI_ENGINE_ID -> KyuubiSparkUtil.engineId)
+    sc.uiWebUrl.foreach { url =>
+      map ++= Map(KYUUBI_ENGINE_UI_URL -> url)
```

Comment:
`WithKyuubiServer` disables spark ui by default (`conf.set("spark.ui.enabled", "false")`) , so UT appears NPE because there is no UI URL.

But after enable spark ui, we will find that some characters in spark ui url are illegal.

org.apache.curator.utils.PathUtils#validatePath
```
            } else if (c == '/' && lastc == '/') {
                reason = "empty node name specified @" + i;
                break;
```

```
Caused by: java.lang.IllegalArgumentException: Invalid path string "/kyuubi_test_1.7.0-SNAPSHOT_USER_SPARK_SQL/csy/default/serviceUri=localhost:56196;version=1.7.0-SNAPSHOT;kyuubi.engine.id=local-1667988857376-692873171;kyuubi.engine.ui.url=http://XXX:56175;refId=f7bae472-5e6e-42e9-8013-a8f72b3adfdd;sequence=" caused by empty node name specified @179
	at org.apache.kyuubi.shade.org.apache.curator.utils.PathUtils.validatePath(PathUtils.java:102)
	at org.apache.kyuubi.shade.org.apache.curator.framework.recipes.nodes.PersistentNode.<init>(PersistentNode.java:153)
	at org.apache.kyuubi.ha.client.zookeeper.ZookeeperDiscoveryClient.createPersistentNode(ZookeeperDiscoveryClient.scala:390)
```